airbyte-integrations/connectors/source-chargebee/source_chargebee/streams.py

```diff
@@ -132,6 +117,77 @@ def read_records(
         # Always return an empty generator just in case no records were ever yielded
         yield from []
 
+
+class SemiIncrementalChargebeeStream(ChargebeeStream):
+    cursor_field = "updated_at"
+
+    def __init__(self, start_date: str):
+        # Convert `start_date` to timestamp(UTC).
+        self._start_date = pendulum.parse(start_date).int_timestamp if start_date else None
+
+    def get_starting_point(self, stream_state: Mapping[str, Any], item_id: str) -> str:
+        start_point = self._start_date
+
+        if stream_state and stream_state.get(item_id, {}).get(self.cursor_field):
+            start_point = max(start_point, stream_state[item_id][self.cursor_field])
+
+        return start_point
+
+    def parse_response(
+        self, list_result: ListResult, stream_slice: Mapping[str, Any] = None, stream_state: Mapping[str, Any] = None
+    ) -> Iterable[Mapping]:
+        starting_point = self.get_starting_point(stream_state=stream_state, item_id=stream_slice["item_id"])
+
+        for message in list_result:
+            record = message._response[self.name]
+            if record[self.cursor_field] > starting_point:
```

Comment:
```suggestion
            if record[self.cursor_field] >= starting_point:
```

Reply:
If we do`>=` we will have duplicates at the beginning of each stream read. For example, if we already have `1627671472` in the state, this means that we are sure that all records with `updated_at <= 1627671472` were already synced.

Reply:
If we do`>=` we will have duplicates at the beginning of each stream read. For example, if we already have `1627671472` in the state, this means that we are sure that all records with `updated_at <= 1627671472` were already synced.